backend/capellacollab/projects/toolmodels/modelsources/git/github/handler.py

```diff
@@ -179,7 +179,9 @@ def __get_latest_successful_job(self, jobs: list, job_name: str) -> dict:
         ):
             raise git_exceptions.GitPipelineJobFailedError(job_name)
 
-        return None
+        raise git_exceptions.GitPipelineJobUnknownStateError(
+            job_name, matched_jobs[0]["conclusion"]
+        )
```

Comment:
Why not make the return type `requests.Response | None`?

Reply:
We have to deal with the case that no previous condition was successful somewhere, so I thought here would be the best place (i.e., when possibly returning `None` we would also have to either raise or return `None` in the `get_last_job_run_id_for_git_model` function and so on).